### PR TITLE
Deprecates builder classes' constructors over builder methods

### PR DESCRIPTION
**Pull request by 4whomtbts**
Motivation:

#1719

This pull request is follow up PR of #2085.
Modifications:

This PR affects below classes's way of obtaining instance of Its builder class 
- `ServerListener`
- `SimpleCompositeService`
- `SimpleCompositeRpcService`
- `CorsPolicy.builder`
- `LoggingService`
- `JettyService`
- `SamlServiceProvider`
- `ZooKeeperUpdatingListener`

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/server/ServerListener.java [hunk 1]
```diff
@@ -23,6 +23,14 @@
  * @see Server#removeListener(ServerListener)
  */
 public interface ServerListener {
+
+    /**
+     * Creates a new {@link ServerListenerBuilder}.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/ServerListener.java]: `Returns a new ...`

File: core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java [hunk 2]
```diff
@@ -68,6 +68,14 @@
      * */
     private final List<Consumer<? super Server>> serverStoppedCallbacks = new ArrayList<>();
 
+    /**
+     * Returns a new {@link ServerListenerBuilder}.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java]: `Creates a new ...`

File: zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListenerBuilder.java [hunk 3]
```diff
@@ -90,7 +93,10 @@ public ZooKeeperUpdatingListenerBuilder(CuratorFramework client, String zNodePat
      *
      * @param connectionStr the ZooKeeper connection string
      * @param zNodePath the ZooKeeper node to register
+     *
+     * @deprecated Use {@link ZooKeeperUpdatingListener#builder(String, String)}
```
Comment (trustin) [inline zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListenerBuilder.java]: Missing period

File: grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcDocServiceTest.java [hunk 4]
```diff
@@ -110,25 +110,24 @@ protected void configure(ServerBuilder sb) throws Exception {
                                        .supportedSerializationFormats(GrpcSerializationFormats.values())
                                        .enableUnframedRequests(true)
                                        .build());
-            sb.serviceUnder(
-                    "/docs/",
-                    new DocServiceBuilder()
-                            .exampleRequestForMethod(
+            sb.serviceUnder("/docs/",
+                    DocService.builder()
```
Comment (minwoox) [inline grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcDocServiceTest.java]: indentation?

File: grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcDocServiceTest.java [hunk 5]
```diff
@@ -110,25 +110,24 @@ protected void configure(ServerBuilder sb) throws Exception {
...
+                              .injectedScript(INJECTED_HEADER_PROVIDER1, INJECTED_HEADER_PROVIDER2)
+                              .injectedScriptSupplier((ctx, req) -> INJECTED_HEADER_PROVIDER3)
+                              .exclude(DocServiceFilter.ofMethodName(TestServiceGrpc.SERVICE_NAME, "EmptyCall"))
+                              .build()
                             .decorate(LoggingService.newDecorator()));
-            sb.serviceUnder("/excludeAll/", new DocServiceBuilder()
-                    .exclude(DocServiceFilter.ofGrpc())
-                    .build());
+            sb.serviceUnder("/excludeAll/", DocService.builder()
+                                                                .exclude(DocServiceFilter.ofGrpc())
```
Comment (minwoox) [inline grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcDocServiceTest.java]: indentation?

File: jetty/src/test/java/com/linecorp/armeria/server/jetty/JettyServiceTest.java [hunk 6]
```diff
@@ -63,21 +63,25 @@ protected void configure(ServerBuilder sb) throws Exception {
...
                     "/jsp/",
-                    new JettyServiceBuilder()
-                            .handler(newWebAppContext())
-                            .configurator(s -> jettyBeans.addAll(s.getBeans()))
-                            .build()
+                    JettyService.builder()
+                                .handler(newWebAppContext())
+                                .configurator(s -> jettyBeans.addAll(s.getBeans()))
+                                .build()
                             .decorate(LoggingService.newDecorator()));
```
Comment (minwoox) [inline jetty/src/test/java/com/linecorp/armeria/server/jetty/JettyServiceTest.java]: indentation?

File: saml/src/test/java/com/linecorp/armeria/server/saml/SamlServiceProviderTest.java [hunk 7]
```diff
@@ -169,37 +169,43 @@
...
-                    .ssoEndpoint(ofHttpRedirect("http://idp.example.com/saml/sso/redirect"))
-                    .sloResEndpoint(ofHttpRedirect("http://idp.example.com/saml/slo/redirect"))
-                    .and()
-                    // We have two IdP config so one of them will be selected by the path variable.
-                    .idpConfigSelector((configurator, ctx, req) -> {
+            final SamlIdentityProviderConfigSelector configSelector =
+                    (configurator, ctx, req) -> {
                         final String idpEntityId = "http://idp.example.com/" +
-                                                   ctx.pathParam("bindingProtocol");
+                                ctx.pathParam("bindingProtocol");
```
Comment (minwoox) [inline saml/src/test/java/com/linecorp/armeria/server/saml/SamlServiceProviderTest.java]: indentation?

File: saml/src/test/java/com/linecorp/armeria/server/saml/SamlServiceProviderTest.java [hunk 8]
```diff
@@ -169,37 +169,43 @@
...
                                 configurator.idpConfigs().get(idpEntityId));
-                    })
-                    .requestIdManager(requestIdManager)
-                    .build();
+                    };
+            final SamlServiceProvider sp =
+                    SamlServiceProvider.builder()
+                                        // A request will be authenticated if it contains 'test=test'
+                                        // cookie in Cookie header.
+                                        .authorizer(new CookieBasedAuthorizer("test", "test"))
```
Comment (minwoox) [inline saml/src/test/java/com/linecorp/armeria/server/saml/SamlServiceProviderTest.java]: indentation. Could you check all the files again please?

Comment (4whomtbts) [reply saml/src/test/java/com/linecorp/armeria/server/saml/SamlServiceProviderTest.java]: Yes. I checked them all again and reflected on new commit.
I'd go over it again and again to catch another this sort of wrong indentation. 
Thanks a lot!

**Comment by trustin** [comment]
Thanks a lot for your help. Saved quite a bit of our time! :bowing_man:

**Comment by 4whomtbts** [comment]
> Thanks a lot for your help. Saved quite a bit of our time! 🙇

Thank you for say that!

**Pull request merged**